tests/RecordTests/FunctionalTest/k4a_cpp_ft.cpp

```diff
@@ -0,0 +1,453 @@
+
+        record recorder4 = std::move(recorder2); // deep copy
+        ASSERT_TRUE(recorder4 != nullptr);
+        ASSERT_FALSE(recorder4 == nullptr);
+
+        recorder2.close();
+        recorder4.close();
+    }
+
+    recorder.add_tag("K4A_CPP_FT_ADD_TAG", "K4A_CPP_FT_ADD_TAG");
+    recorder.add_imu_track();
+
+    std::string k4a_cpp_ft_attachment = "K4A_CPP_FT_ADD_ATTACHMENT";
+    recorder.add_attachment(k4a_cpp_ft_attachment.c_str(),
+                            (uint8_t *)k4a_cpp_ft_attachment.data(),
+                            k4a_cpp_ft_attachment.size());
+
+    k4a_record_video_settings_t vid_settings = { 1920, 1080, 30 };
+    std::string k4a_cpp_ft_custom_vid_track = "K4A_CPP_FT_CUSTOM_VID_TRACK";
+    recorder.add_custom_video_track(k4a_cpp_ft_custom_vid_track.c_str(), "V_MPEG1", nullptr, 0, &vid_settings);
+
+    k4a_record_subtitle_settings_t st_track = { false };
+    std::string k4a_cpp_ft_custom_subtitle_track = "CUSTOM_K4A_SUBTITLE_TRACE";
+    recorder.add_custom_subtitle_track(k4a_cpp_ft_custom_subtitle_track.c_str(), "V_MPEG1", nullptr, 0, &st_track);
+
+    recorder.write_header();
+
+    for (int x = 0; x < 100; x++)
+    {
+        capture capture;
+        k4a_imu_sample_t imu;
+        if (kinect.get_capture(&capture, std::chrono::milliseconds(1000)))
+        {
+            recorder.write_capture(capture);
+        }
+        while (kinect.get_imu_sample(&imu, std::chrono::milliseconds(0)))
+        {
+            recorder.write_imu_sample(imu);
+        }
+        image color = capture.get_color_image();
+        recorder.write_custom_track_data(k4a_cpp_ft_custom_vid_track.c_str(),
+                                         color.get_device_timestamp(),
+                                         color.get_buffer(),
+                                         color.get_size());
+        color.reset();
+
+        image depth = capture.get_depth_image();
+        recorder.write_custom_track_data(k4a_cpp_ft_custom_subtitle_track.c_str(),
+                                         depth.get_device_timestamp(),
+                                         depth.get_buffer(),
+                                         depth.get_size());
+        capture.reset();
+    }
+    recorder.flush();
+    kinect.stop_cameras();
+    kinect.stop_imu();
+}
+
+TEST_F(k4a_cpp_ft, playback)
+{
+    playback pb = playback::open("./k4a_cpp_ft.mkv");
+    playback pb_empty;
+    ASSERT_TRUE(pb);             // bool operation
+    ASSERT_TRUE(pb != nullptr);  //!= nullptr
+    ASSERT_FALSE(pb == nullptr); // == nullptr
+    ASSERT_TRUE(pb != pb_empty);
+    ASSERT_FALSE(pb == pb_empty);
+
+    // Deep copy
+    playback pback = std::move(pb);
+    ASSERT_TRUE(pback != nullptr);
+    ASSERT_TRUE(pb == nullptr);
+
+    std::vector<uint8_t> raw_cal = pback.get_raw_calibration();
+    std::cout << "calibration is : ";
+    for (std::vector<uint8_t>::const_iterator i = raw_cal.begin(); i != raw_cal.end(); ++i)
```

Comment:
>std::vector<uint8_t>::const_iterator i = raw_cal.begin(); i != raw_cal.end(); ++i [](start = 9, length = 81)

This is fine, but FYI C++ has foreach loops as of C++11 - you can write this more concisely as:
for (const uint8_t& char : raw_cal)
{
    std::cout << char;
}

Reply:
Thanks!